tests/test_agent_manager.py

```diff
@@ -307,13 +349,39 @@ async def test_api(init_dataclasses_and_load_schema):
                 ],
                 "environment": env4.id,
             },
+            {
+                "first_seen": UNKWN,
+                "expired": expiration,
+                "hostname": "ts5",
+                "last_seen": UNKWN,
+                "endpoints": [
+                    {"expired": expiration, "id": UNKWN, "name": "agent1", "process": UNKWN},
+                ],
+                "environment": env5.id,
+            },
         ]
     }
 
-    assert_equal_ish(shouldbe, all_agents, sortby=["hostname", "name"])
-    agentid = all_agents["processes"][0]["sid"]
+    assert_equal_ish(shouldbe, all_agents_processes, sortby=["hostname", "name"])
+    agentid = all_agents_processes["processes"][0]["sid"]
+
+    start = agentid
+    code, all_agents_processes = await am.list_agent_processes(environment=None, expired=True, start=start)
+    assert code == 200
+    for agent_process in all_agents_processes["processes"]:
+        assert (
+            agent_process["sid"] > start
+        ), f"List of agent processes should not contain a sid (={agent_process['sid']}) before or equal to start (={start})"
+
+    end = agentid
+    code, all_agents_processes = await am.list_agent_processes(environment=None, expired=True, end=end)
+    assert code == 200
+    for agent_process in all_agents_processes["processes"]:
+        assert (
+            agent_process["sid"] < end
+        ), f"List of agent processes should not contain a sid (={agent_process['sid']}) after or equal to end (={end})"
```

Comment:
Also assert the number of records here. If the result is an empty list, this test will always succeed.

Reply:
Did it in https://github.com/inmanta/inmanta/pull/2521/commits/3ef9c64b173a26c0fa2feebd57a595f17006ed2b